src/renderer/PresetsBar.tsx

```diff
@@ -84,48 +90,51 @@ const PresetListItem = ({
 };
 
 const PresetsBar = () => {
-  const { globalError } = useAquaContext();
+  const { globalError, performHealthCheck, setGlobalError } = useAquaContext();
 
   const [presetName, setPresetName] = useState<string>('');
   const [selectedPresetName, setSelectedPresetName] = useState<
     string | undefined
   >(undefined);
 
-  // TODO: Fetch default presets and custom presets from storage
-  const [presetNames, setPresetNames] = useState<string[]>([
-    'Airpods Pro to IEF Neutral AutoEQ',
-    'Airpods to Harman AE OE 2018 AutoEQ',
-    '7-1 Surround Sound Speaker Setup',
-    'Bass Boost',
-    'Bass and Treble Boost',
-    'Classic',
-    'Dance',
-    'Rock',
-    'Vocal Boost',
-    'Microphone Compensation',
-    'MORE BASS',
-    'Surround effect',
-    'Indie playlist',
-    'Pop',
-    'Hip-hop',
-    'R&B',
-    'Equalize for shrill noise',
-    'Jazz',
-  ]);
-
-  const handleCreatePreset = (prev: string[]) => {
-    if (prev.indexOf(presetName) !== -1) {
-      // TODO: overwrite preset values for the existing preset
-    } else {
-      // TODO: add code for creating a preset
+  const [presetNames, setPresetNames] = useState<string[]>([]);
+
+  // Fetch default presets and custom presets from storage
+  useEffect(() => {
+    const fetchPresetNames = async () => {
+      const result = await getPresetListFromFiles();
+      // Sort preset names before updating state just in case
+      setPresetNames(result.sort());
+    };
+
+    fetchPresetNames();
+  }, []);
+
+  const handleCreatePreset = async (prev: string[]) => {
+    await savePreset(presetName);
```

Comment:
exception case

Reply:
same, not sure what to do here. just continue running ig?